returnn/datasets/postprocessing.py

```diff
@@ -188,7 +194,7 @@ def init_seq_order(
         self._dataset.init_seq_order(epoch=epoch, seq_list=seq_list, seq_order=seq_order)
         self._data_iter = enumerate(self._build_mapping_iter())
         self._seq_list_for_validation = seq_list
-        if self._map_seq_stream is None:
+        if self._map_seq_stream is None or self._iterator_preserves_sequences:
             # If we don't have an iterable mapper we know the number of segments exactly
             # equals the number of segments in the wrapped dataset
```

Comment:
Ok, if you want to add this functionality then this comment should also be adapted :)